google-cloud-firestore/lib/google/cloud/firestore/commit_response.rb

```diff
@@ -0,0 +1,139 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/firestore/v1beta1"
+require "google/cloud/firestore/convert"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # CommitResponse
+      #
+      # The response for a commit.
+      #
+      # @example
+      #   require "google/cloud/firestore"
+      #
+      #   firestore = Google::Cloud::Firestore.new
+      #
+      #   commit_response = firestore.batch do |b|
+      #     # Set the data for NYC
+      #     b.set("cities/NYC", { name: "New York City" })
+      #
+      #     # Update the population for SF
+      #     b.update("cities/SF", { population: 1000000 })
+      #
+      #     # Delete LA
+      #     b.delete("cities/LA")
+      #   end
+      #
+      #   puts commit_response.commit_time
+      #   commit_response.write_results.each do |write_result|
+      #     puts write_result.update_time
+      #     puts write_result.transform_results.inspect
+      #   end
+      #
+      class CommitResponse
+        ##
+        # The time at which the commit occurred.
+        #
+        # @return [Time] The commit time.
+        attr_accessor :commit_time
+
+        ##
+        # The result of applying the writes.
+        #
+        # This i-th write result corresponds to the i-th write in the request.
+        #
+        # @return [Array<CommitResponse::WriteResult>] The write results.
+        attr_accessor :write_results
+
+        ##
+        # @private
+        def self.from_grpc grpc, client
+          commit_time = Convert.timestamp_to_time grpc.commit_time
+          write_results = Array(grpc.write_results).map do |write_result|
+            WriteResult.from_grpc write_result, client
```

Comment:
Can you file a GitHub issue (or add a TODO) to strip WriteResults for DocumentTransforms?

Reply:
Yes, I'll plan on addressing this in a separate PR. But I have many questions about what these requirements are.